scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -211,7 +211,7 @@ class ScioContext private[scio] (val options: DataflowPipelineOptions,
       throw new RuntimeException("Cannot set name once pipeline is initialized")
     }
     // override app name and job name
-    options.setAppName(name)
+    if (options.getAppName == null) options.setAppName(name)
```

Comment:
But this will make `setName` a no-op if app name is already set.

`setName` is actually used only once in `ScioContext` constructor, so maybe do the `== null` check there?
https://github.com/rouzwawi/scio/blob/master/scio-core/src/main/scala/com/spotify/scio/ScioContext.scala#L129